driver/src/main/native/com/code_intelligence/jazzer/driver/fuzzed_data_provider.cpp

```diff
@@ -131,15 +121,21 @@ template <typename T>
 T JNICALL ConsumeIntegralInRange(JNIEnv &env, jobject self, T min, T max) {
   uint64_t range = static_cast<uint64_t>(max) - min;
   uint64_t result = 0;
-  std::size_t offset = 0;
+  jint offset = 0;
+
+  const auto *dataPtr =
+      reinterpret_cast<const uint8_t *>(env.GetLongField(self, gDataPtrField));
+  jint remainingBytes = env.GetIntField(self, gRemainingBytesField);
 
   while (offset < 8 * sizeof(T) && (range >> offset) > 0 &&
-         gRemainingBytes != 0) {
-    --gRemainingBytes;
-    result = (result << 8u) | gDataPtr[gRemainingBytes];
+         remainingBytes != 0) {
+    --remainingBytes;
+    result = (result << 8u) | dataPtr[remainingBytes];
     offset += 8;
   }
 
+  env.SetIntField(self, gRemainingBytesField, remainingBytes);
```

Comment:
I think I don't fully understand what's happening here. Don't we have to set the new position in `gDataPtrField` as well?

Reply:
The loop above only reads from `dataPtr`, but doesn't modify it, so we don't have to update the field. I will add a comment.

Reply:
So we take the bytes we need from the end of the array and just modify `remainingBytes` accordingly? But why is this handled differently to the other methods?

Reply:
Bulk data is taken from the front, bytes used to generate short primitive types are taken from the end. The idea is that of you are fuzzing a jpeg parser, your Corpus entries should start with jpeg data even if you consume some additional bytes to e.g. select parser settings. The original FuzzedDataProvider did this and I just copied the logic